state_machine.h: fix initialization of state_, chg_time_

xqms opened the pull request:
The old code called operator=() to initialize, which checked
state_ to determine whether a change was really needed. Of course,
at that point in time state_ was still uninitialized (=> undefined behavior).
So do the initialization explicitly.

Found using valgrind.

dgossow commented:
Thanks for fixing this. One stylistic nit though: the variables should be initialized in the initializer list, like name_. Would you mind changing that?

xqms commented:
I agree, that's better. I pushed a new version...

xqms committed:
state_machine.h: fix initialization of state_, chg_time_

The old code called operator=() to initialize, which checked
state_ to determine whether a change was really needed. Of course,
at that point in time state_ was still uninitialized. So do the
initialization explicitly.

Found using valgrind.

dgossow commented:
Thanks!
